cardano-explorer/src/Explorer/Web/Query.hs

```diff
@@ -127,3 +128,47 @@ queryTx hash = do
             where_ (tx ^. TxHash ==. val hash)
             pure tx
   pure $ maybeToEither (DbLookupTxHash hash) entityPair (listToMaybe res)
+
+data TxWithInputsOutputs = TxWithInputsOutputs
+  { txwTx :: Tx
+  , txwInputs :: [(Text, Word64)]
+  , txwOutputs :: [TxOut]
+  }
+
+queryBlockTxs :: MonadIO m => ByteString -> Int64 -> Int64 -> ReaderT SqlBackend m ([ TxWithInputsOutputs ], Word64)
+queryBlockTxs blkHash limitNum offsetNum = do
+  let
+    blockid = subList_select . from $ \blk -> do
```

Comment:
Do you think using `where` causes might make this clearer?

`where` clauses allow you to put the meat of the function close to the type signature and helper functions in the `where.